src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+import {
+  Component,
+  Element,
+  Host,
+  h,
+  Prop,
+  VNode,
+  Event,
+  EventEmitter,
+  Watch,
+  State,
+  Listen,
+  Method
+} from "@stencil/core";
+import { Scale, Theme } from "../interfaces";
+import { AmPm, formatNumberAsTimeString, MinuteOrSecond, numberKeys, Time } from "./utils";
+
+@Component({
+  tag: "calcite-time-picker",
+  styleUrl: "calcite-time-picker.scss",
+  shadow: true
+})
+export class CalciteTimePicker {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteTimePickerElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** The hour value (24-hour format) */
+  @Prop({ reflect: true, mutable: true }) hour?: string = "--";
```

Comment:
Do we need the default values to be `--`? Seems like `undefined` would be better for checking if the value is not set.

Reply:
Refactored this out in https://github.com/Esri/calcite-components/pull/1736/commits/a388267fe49cc991081e1cdd960e5f0801fd0b48